### PR TITLE
Allow to add services to plugins

### PR DESCRIPTION
**Pull request by erikzhang**
This PR allows the client to add services to the plugin system. For example, we can add `ConsoleService` in neo-cli to the plugin system to allow the plugins to access to the console.

### CODE REVIEW

File: src/neo/Plugins/Plugin.cs [hunk 1]
```diff
@@ -19,6 +20,7 @@ public abstract class Plugin : IDisposable
         internal static readonly List<IP2PPlugin> P2PPlugins = new List<IP2PPlugin>();
         internal static readonly List<IMemoryPoolTxObserverPlugin> TxObserverPlugins = new List<IMemoryPoolTxObserverPlugin>();
 
+        private static ImmutableList<object> services = ImmutableList<object>.Empty;
```
Comment (shargon) [inline src/neo/Plugins/Plugin.cs:23]:
```suggestion
        private static readonly ImmutableList<object> services = ImmutableList<object>.Empty;
```

Comment (erikzhang) [reply src/neo/Plugins/Plugin.cs:23]: The `ImmutableList` can't be `readonly`.

**Review by shargon — APPROVED** [review]

**Comment by Tommo-L** [comment]
but we should define `ConsoleService` interface in neo-core

**Comment by erikzhang** [comment]
> but we should define ConsoleService interface in neo-core

We don't need to. The plugin need to refer to the `Neo.ConsoleService` package.

**Pull request merged**